lucet-spectest/src/script.rs

```diff
@@ -67,8 +67,14 @@ impl ScriptEnv {
     }
     pub fn instantiate(&mut self, module: &[u8], name: &Option<String>) -> Result<(), ScriptError> {
         let bindings = bindings::spec_test_bindings();
-        let compiler = Compiler::new(module, OptLevel::Fast, &bindings, HeapSettings::default())
-            .map_err(program_error)?;
+        let compiler = Compiler::new(
+            module,
+            OptLevel::Fast,
+            &bindings,
+            HeapSettings::default(),
+            false,
```

Comment:
I flipped this to true because why not, and it turns out this as-is was causing a lot of spectest issues.

There's a lot of dead code in the spec tests that also cause the instrumentation logic to add code where Cranelift expects there to be no more. I've added some additional logic to figure out when wasm ops are dead code, and the spec tests all pass now, but it also pretty much duplicates the dead code logic out of Cranelift. Passing along the cranelift-wasm translation state exposes a lot of translation innards and really doesn't make for a great interface..